src/hivpy/sexual_behaviour.py

```diff
@@ -277,3 +284,99 @@ def update_rred_balance(self, population):
         else:
             population.loc[men, col.RRED_BALANCE] = 1/rred_balance
             population.loc[women, col.RRED_BALANCE] = rred_balance
+
+    def update_ltp_rate_change(self, date):
+        if date1995 < date < date2000:
+            dt = diff_years(date1995, date)
+            self.ltp_rate_change = self.annual_ltp_rate_change**(dt)
+        elif date >= date2000:
+            self.ltp_rate_change = self.annual_ltp_rate_change**5
+            # don't really need to keep doing this update after 2000
+
+    def balance_ltp_factor(self, population):
+        num_ltp_women = sum(
+            population.data.loc[population.data[col.SEX] == SexType.Female, col.LONG_TERM_PARTNER])
+        num_ltp_men = sum(
+            population.data.loc[population.data[col.SEX] == SexType.Male, col.LONG_TERM_PARTNER])
+        if num_ltp_women > 0:
+            ratio_ltp = num_ltp_men / num_ltp_women
+            if ratio_ltp < 0.8:
+                return (4, 1)
+            elif ratio_ltp < 0.9:
+                return (2, 1)
+            elif ratio_ltp < 1.1:
+                return (1, 1)
+            elif ratio_ltp < 1.2:
+                return (1, 2)
+            else:
+                return (1, 4)
+        else:
+            return (1, 1)
+
+    def start_new_ltp(self, sex, age, size):
+        # TODO add alterations for new diagnosis
+        ltp_rands = rng.random(size=size)
+        rate_modifier = 1
+        if (35 <= age < 45):
+            rate_modifier = 0.5
+        elif (45 <= age < 55):
+            rate_modifier = 1/3
+        else:
+            rate_modifier = 0.2
+        new_relationship = ltp_rands < (self.new_ltp_rate
+                                        * rate_modifier
+                                        * self.ltp_balance_factor[sex])
+        return new_relationship
+
+    def new_ltp_longevity(self, age, sex, size):
+        longevity = np.zeros(size)  # each new relationship needs a longevity
+        longevity_rands = rng.random(size)
+        if (age < 45):
+            longevity = ((longevity_rands < 0.3) * 1 +
+                         ((0.3 <= longevity_rands) & (longevity_rands < 0.6)) * 2 +
+                         (0.6 <= longevity_rands) * 3)
```

Comment:
My initial thought about writing this would be with `.loc` - any benefits to either?
  ```suggestion
              longevity.loc[longevity_rands < 0.3, :] = 1
              longevity.loc[(0.3 <= longevity_rands) & (longevity_rands < 0.6), :] = 2
              longevity.loc[0.6 <= longevity_rands, :] = 3
```

Reply:
I also wonder if combining the code for different age categories makes sense, e.g.
```python
cat2_threshold = 0.6 if age < 45 else 0.8 if age < 55 else 1.1
...
longevity.loc[(0.3 <= longevity_rands) & (longevity_rands < cat2_threshold), :] = 2
```
(or with the sum approach as is)

Reply:
Oh, sorry, `longevity` is a numpy array, not a panda object. We can still assign a subset of elements with a logical index though:
```python
longevity[longevity_rands < 0.3] = 1
```